src/backends/beamcoder.ts

```diff
@@ -365,6 +367,47 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
         return this.getFrameAtPts(targetPts);
     }
 
+    /**
+     * Dump one frame's image data (for canvas) at a specific time
+     *
+     * This method can seek as required, but generally, it is designed to be
+     * performant in the cases where we progressively read a video frame by frame.
+     *
+     * So the implementation in here should not seek all the time, but rather
+     * read packets as they come, when possible,
+     *
+     * @see getFrameAtPts()
+     */
+    async getFrameImageDataAtTime(targetTime: number): Promise<ImageData> {
+        LOG_SINGLE_FRAME_DUMP_FLOW && console.log(`Requesting to dump a frame at Time(s)=${targetTime}`);
+        const targetPts = Math.floor(this.timeToPts(targetTime));
+        const frame = await this.getFrameAtPts(targetPts);
+
+        const components = 4; // 4 components: r, g, b and a
```

Comment:
We should assert that outputPixelFormat is 'rgba' here

Reply:
I guess we shouldn't be able to specify this right now since `imageData` always requires `rgba` @antoineMoPa?

Reply:
Yeah, maybe we can simplify that!